scripts/doc_checker.py

```diff
@@ -0,0 +1,546 @@
+    """
+    is_successfull = True
+
+    if path.is_file():
+        walker = ((str(path.parent), [], [path.name]),)
+    else:
+        walker = os.walk(path)
+
+    for root, _, files in walker:
+        if "__pycache__" in root:
+            continue
+        for f in files:
+            if not f.endswith(".py"):
+                continue
+
+            current_path = os.path.join(root, f)
+            # get importable name
+            module_name = current_path.replace("/", ".").replace("\\", ".")
+            module_name = module_name.replace(".py", "")
+
+            with open(current_path) as fd:
+                file_contents = fd.read()
+
+            # using static parsing for collecting module, functions, classes and their methods
+            module = ast.parse(file_contents)
+
+            def is_public_func(node):
+                return isinstance(node, ast.FunctionDef) and (
+                    not node.name.startswith("__") or node.name.endswith("__")
+                )
+
+            functions = [node for node in module.body if is_public_func(node)]
+            classes = [node for node in module.body if isinstance(node, ast.ClassDef)]
+            methods = [
+                f"{module_name}.{_class.name}.{node.name}"
+                for _class in classes
+                for node in _class.body
+                if is_public_func(node)
+            ]
+
+            logging.info(f"NUMPYDOC OUTPUT FOR {current_path} - CAN BE EMPTY")
+            # numpydoc docstrings validation
+            # docstrings are taken dynamically
+            to_validate = (
+                [module_name]
+                + [f"{module_name}.{x.name}" for x in (functions + classes)]
+                + methods
+            )
+            if not all(list(map(validate_object, to_validate))):
+                is_successfull = False
+    return is_successfull
+
+
+def pydocstyle_validate(path: pathlib.Path, add_ignore: List[str]) -> int:
+    """
+    Perform pydocstyle checks.
+
+    Parameters
+    ----------
+    path : pathlib.Path
+        Filename or directory path for check.
+    add_ignore : List[int]
+        `pydocstyle` error codes which are not verified.
+
+    Returns
+    -------
+    bool
+        Return True if all pydocstyle checks are successful.
+    """
+    pydocstyle = "pydocstyle"
+    if not shutil.which(pydocstyle):
+        raise ValueError(f"{pydocstyle} not found in PATH")
+    result = subprocess.run(
+        [
+            pydocstyle,
+            "--convention",
+            "numpy",
+            "--add-ignore",
+            ",".join(add_ignore),
+            str(path),
+        ],
+        text=True,
+        capture_output=True,
+    )
+    if result.returncode:
+        logging.info(f"PYDOCSTYLE OUTPUT FOR {path}")
+        logging.error(result.stdout)
+    return True if result.returncode == 0 else False
+
+
+def monkeypatching():
+    """Monkeypatch decorators which change __doc__ attribute."""
```

Comment:
Doesn't `GL01` say that a docstring should start in the line below?

Reply:
The check is only used for multi-line docstring.

Reply:
Okay, I see.